sounddevice.py

```diff
@@ -2262,13 +2262,68 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
```

Comment:
OK, I wasn't thinking about that. I don't know if anyone will use this, but I guess there is no reason for not making that possible. There are no disadvantages from our side, right?

But that would mean you'd have to move the initialization code out of the constructor again ...

I guess we could change the constructor to:

```python
__init__(self, msg='', code=None, hosterror_info=None)
```

... and just do all the PortAudio calls outside of it in `_check()`.

Then it should be easy to re-raise with a new message:

```python
try:
    # something
except sd.PortAudioError as e:
    raise sd.PortAudioError('Ayayayay!', e.code, e.hosterror_info) from e
```

Were you thinking about something like that?